cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,710 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+import warnings
+from typing import List, Dict, Any, Optional, Tuple
+import numpy as np
+import copy
+
+# for visualizing functions
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
```

Comment:
all matplotlib imports need to be lazy imports such that this is an optional dependency of the package. Do not add this dependency to our optional dependencies for now, but remember that it is not mentioned. The datalab PR will re-organize the dependencies into optional ones, so after it is merged, you can just add matplotlib to the list of optional dependencies.